src/Algebra/Graph.hs

```diff
@@ -1285,29 +1288,37 @@ sparsifyKL n graph = KL.buildG (1, next - 1) ((n + 1, n + 2) : Exts.toList (res
 The rules for foldg work very similarly to GHC's mapFB rules; see a note below
 this line: http://hackage.haskell.org/package/base/docs/src/GHC.Base.html#mapFB.
 
-* Up to (but not including) phase 1, we use the "buildR/f" rule to rewrite all
-  saturated applications of f into its buildR/foldg form, hoping for fusion to
-  happen (through the "foldg/buildR" rule).
+* Up to (but not including) phase 1, we use the "buildg/f" rule to rewrite all
+  saturated applications of f into its buildg/foldg form, hoping for fusion to
+  happen (through the "foldg/buildg" rule).
 
-  In phases 1 and 0, we switch off these rules, inline buildR, and switch on the
+  In phases 1 and 0, we switch off these rules, inline buildg, and switch on the
   "graph/f" rule, which rewrites "foldg/f" back into plain functions if needed.
 
   It's important that these two rules aren't both active at once (along with
   build's unfolding) else we'd get an infinite loop in the rules. Hence the
   activation control below.
 
 * composeR and matchR are here to remember the original function after applying
-  a "buildR/f" rule. These functions are higher-order functions and therefore
+  a "buildg/f" rule. These functions are higher-order functions and therefore
   benefit from inlining in the final phase.
 
 * The "bindR/bindR" rule optimises compositions of multiple bindR's.
 -}
 
+-- | A function abstracting graphs constructors.
```

Comment:
I think it's worth providing some more explanation. For example, pointing out that it's actually the type of `foldg`, right? I guess it is also related dto the Boehm-Berarducci encoding of algebraic graphs:

https://hackage.haskell.org/package/algebraic-graphs-0.4/docs/Algebra-Graph-Fold.html

Note, however, that `Fold` will be deleted in the next release... Maybe I shouldn't have done that :)

Reply:
> I guess it is also related dto the Boehm-Berarducci encoding of algebraic graphs

Ah, yes indeed. If you are going to remove it, I don't know if it worth adding a reference.

I am not fond of "this is foldg's type", because, for me, it hides the use-case of building a structure (ie, one can express `vertices` with `buildg`).